ivy/functional/backends/torch/experimental/manipulation.py

```diff
@@ -435,7 +435,7 @@ def unique_consecutive(
 
 def fill_diagonal(
     a: torch.Tensor,
```

Comment:
let's remove the torch implementation as well since there isn't any a native `fill_diagonal` function that we are leveraging here. Our compositional implementation in ivy would be just as efficient when the code is compiled.